src/pages/en/core-concepts/endpoints.md

```diff
@@ -0,0 +1,275 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Endpoints
+description: Learn how to create endpoints that serve any kind of data
+i18nReady: true
+---
+Astro lets you create custom endpoints to serve any kind of data. You can use this to generate images, expose an RSS document, or build a full API for your site.
+
+If your project is in [static](/en/reference/configuration-reference/#output) mode, custom endpoints are called at build time to produce static files. In [SSR](/en/guides/server-side-rendering/) mode, custom endpoints turn into live server endpoints that are called on request. Static and SSR endpoints are defined similarly, but SSR endpoints support additional features.
+
+## Static File Endpoints
+To create a custom endpoint, add a `.js` or `.ts` file to the `/pages` directory. The `.js` or `.ts` extension will be removed during the build process, so the name of the file should include the extension of the data you want to create. For example, `src/pages/data.json.ts` will build a `/data.json` endpoint.
+
+In that file, export a `get` function (optionally `async`) that returns an object with a `body`. In static mode, Astro will call this at build time to generate the corresponding file.
+
+```js
+// Example: src/pages/builtwith.json.ts
+// Outputs: /builtwith.json
+
+// File routes export a get() function, which gets called to generate the file.
+// Return an object with `body` to save the file contents in your final build.
+export async function get({params, request}) {
+  return {
+    body: JSON.stringify({
+      name: 'Astro',
+      url: 'https://astro.build/',
+    }),
+  };
+}
+```
```

Comment:
``````suggestion
File routes export a `get` function (optionally `async`) which gets called to generate the file. It returns an object with a `body` to save the file contents in your final build. In static mode, Astro will call this at build time to generate the corresponding file.

```js
// Example: src/pages/builtwith.json.ts
// Outputs: /builtwith.json

export async function get({params, request}) {
  return {
    body: JSON.stringify({
      name: 'Astro',
      url: 'https://astro.build/',
    }),
  };
}
```
There's some duplication there to be edited, but I just moved the code comment out into the paragraph itself. I feel like both might have been redundant? Can that literally come out and become part of the lead-in paragraph?

Or was your thinking, "This is for the people who aren't going to read the paragraph, so I'd better put the exact same thing in the code sample?" (I get that! Just does seem to say the same thing twice?)
``````